News entries: Pkg.precompile and auto-precompillation, time macro comp time, varinfo() extended func (#38220)
JuliaLang/julia: NEWS.md

Standard library changes
------------------------

> * `pkg> precompile` is now parallelized, and will run automatically after any `Pkg` action that changes the
>   active manifest. Errors will only throw for direct dependencies listed in the `Project.toml`, and auto-precompillation
>   will remember if a package has errored within the given environment and not retry until it changes. Disable
>   auto-precompillation with `ENV["JULIA_PKG_PRECOMPILE_AUTO"]=0`.

Reviewer: It might be clearer to split into two items and remove unnecessary details.

```suggestion
* `pkg> precompile` now runs in parallel.
* `pkg> precompile` now gets triggered automatically when the active manifest is changed. The auto-precompilation can be disabled by setting environment variable `ENV["JULIA_PKG_PRECOMPILE_AUTO"]=0`.
```

Reviewer 2: >`pkg> precompile` now runs in parallel.

this makes it sound like it runs along-side something else, which isn't the case

The second wording is  ok, but I want to  include the error handling note

```suggestion
* `pkg> precompile` is now parallelized through depth-first precompillation of dependencies. Errors will only throw for direct dependencies listed in the `Project.toml`.
* `pkg> precompile` is now automatically triggered whenever Pkg changes the active manifest. Auto-precompillation will remember if a package has errored within the given environment and not retry until it changes. Disable auto-precompillation by setting the environment variable `JULIA_PKG_PRECOMPILE_AUTO=0`.
```

Reviewer: My argument is that these details are not what normal Julia users would want to concern about. These should be code comments instead of news.

Reviewer 2: Thanks for the feedback. I've tried to improve it a little. 

There is some detail I wanted to take the opportunity to disseminate, but I'm open to reducing it

Standard library changes
------------------------

> * `pkg> precompile` is now parallelized through depth-first precompillation of dependencies. Errors will only throw for

Reviewer 3: ```suggestion
* `pkg> precompile` is now parallelized through depth-first precompilation of dependencies. Errors will only throw for
```

  direct dependencies listed in the `Project.toml`.
* `pkg> precompile` is now automatically triggered whenever Pkg changes the active manifest. Auto-precompillation will
  remember if a package has errored within the given environment and will not retry until it changes.

>   Auto-precompillation can be disabled by setting the environment variable `JULIA_PKG_PRECOMPILE_AUTO=0`.

Reviewer 3: ```suggestion
  Auto-precompilation can be disabled by setting the environment variable `JULIA_PKG_PRECOMPILE_AUTO=0`.
```

* The `Pkg.Artifacts` module has been imported as a separate standard library.  It is still available as
  `Pkg.Artifacts`, however starting from Julia v1.6+, packages may import simply `Artifacts` without importing
  all of `Pkg` alongside. ([#37320])

> * `@time` now reports if the time presented included any compilation time, as a percentage ([#37678])

Reviewer: Hmmm.. I'm not sure of it.

```suggestion
* `@time` now reports the compilation time, if included, as a percentage ([#37678])
```

Reviewer 4: Maybe 
```suggestion
* `@time` now reports compilation time (if any) as a percentage ([#37678])
```

Reviewer 2: Thanks but the original wording is more accurate. It doesn't report total compile time, just any compile time that has contributed to the time reported. For instance, inference time isn't included

Reviewer 2: I think I made it a bit more readable 🤞
